harness/determined/pytorch/_callback.py

```diff
@@ -72,3 +72,27 @@ def load_state_dict(self, state_dict: Dict[str, Any]) -> None:
         Load the state of this using the deserialized ``state_dict``.
         """
         pass
+
+    def on_train_epoch_start(self) -> None:
+        """
+        Run on start of a new training epoch
+        """
+        pass
+
+    def on_train_epoch_end(self, output: List[Any]) -> None:
+        """
+        Run after a new training epoch has finished
+        """
+        pass
+
+    def on_validation_epoch_start(self) -> None:
```

Comment:
Use `on_validate_epoch_start` to be consistent with the other names?

Reply:
hmm what are you suggesting to be consistent with? our other callbacks start with `on_validation`
```
hrz ➜ determined git:(ptl-2) ag on_valida* harness/determined/pytorch/_callback.py                                                                                                   
17:        (except for :meth:`on_validation_end`, :meth:`on_validation_step_end` and
22:    def on_validation_start(self) -> None:
28:    def on_validation_end(self, metrics: Dict[str, Any]) -> None:
37:    def on_validation_step_start(self) -> None:
44:    def on_validation_step_end(self, metrics: Dict[str, Any]) -> None:
88:    def on_validation_epoch_start(self) -> None:
94:    def on_validation_epoch_end(self, outputs: List[Any]) -> None:
```

Reply:
or on.train to on.training

Reply:
I updated ontrain to be consistent with the rest. let's continue in slack with @ahj-determined